control/freqplot.py

```diff
@@ -1160,7 +1160,7 @@ def singular_values_plot(syslist, omega=None,
         else:
             nyquistfrq = None
 
-        fresp = sys(1j*omega if sys.isctime() else np.exp(1j * omega * sys.dt))
+        fresp = sys(1j*omega if sys.isctime() else np.exp(1j * omega * sys.dt)).reshape(sys.noutputs, sys.ninputs, len(omega))
```

Comment:
No. Please make sys return the correct shape by `squeeze=False` and `s` or `z` being a 1D array, which will work around #594 until #595 fixes it.

Reply:
Fixed according to the suggestion, thanks @bnavigator! 

BTW, I was wondering about the return shape of singular_value_plot:

>>> sigma_res, omega_res = ct.singular_values_plot(rss(2,2,2), np.linspace(-3, 2, 1000))
>>> sigma_res.shape
(1000, 2) # omega, singular_value

Is it OK? The general philosophy of python-control seems to be to have "channels first", both for time- and
frequency-domain functions. Perhaps I should transpose the result, then!

Reply:
Yes it should be consistent with `bode_plot()` and thus:

- with `frequency_response()`:
  ```
        Returns
        -------
        mag : ndarray
            The magnitude (absolute value, not dB or log10) of the system
            frequency response.  If the system is SISO and squeeze is not
            True, the array is 1D, indexed by frequency.  If the system is not
            SISO or squeeze is False, the array is 3D, indexed by the output,
            input, and frequency.  If ``squeeze`` is True then
            single-dimensional axes are removed.
   ```

- with `TransferFunction.__call__()` and `StateSpace.__call__()`:
  ```
        Returns
        -------
        fresp : complex ndarray
            The frequency response of the system.  If the system is SISO and
            squeeze is not True, the shape of the array matches the shape of
            omega.  If the system is not SISO or squeeze is False, the first
            two dimensions of the array are indices for the output and input
            and the remaining dimensions match omega.  If ``squeeze`` is True
            then single-dimensional axes are removed.
  ```

Reply:
Of course SISO does not really make sense for SVD.

Reply:
Then I should definitely transpose the result before returning the result!
sigma = sigma.transpose()

Do you care about returning a c-contiguous result? In case I should  do something like
sigma = np.ascontiguousarray(sigma.transpose())

Reply:
> Do you care about returning a c-contiguous result? In case I should do something like
> sigma = np.ascontiguousarray(sigma.transpose())

I don't think that is necessary.

Reply:
OK tks! 

BTW my function does accept SISO input, you get back as SVD the absolute value (aka magnitude).